server/market/market.go

```diff
@@ -385,11 +550,14 @@ func (m *Market) processEpoch(epoch *EpochQueue) {
 	if len(matches) > 0 {
 		log.Debugf("Negotiating %d matches for epoch %d:%d", len(matches),
 			epoch.Epoch, epoch.Duration)
+		// Since Swapper.Negotiate is called asynchronously, we must lock coins
+		// with the Swapper first. The swapper will unlock the coins.
+		m.swapper.LockOrdersCoins(swapOrders)
```

Comment:
I'm not sure I follow. Why can't the Swapper lock the coins asynchronously?

Reply:
I believe what could happen is a new order could arrive that is backed by coins that should be locked by the swapper, but since execution of `go m.swapper.Negotiate` is at the mercy by the Go scheduler, the new order may get accepted if the locking happens inside `Negotiate`.